lib/internal/crypto/random.js

```diff
@@ -281,10 +281,53 @@ function getRandomValues(data) {
   return data;
 }
 
+const hexDigits = [
+  48, 49, 50, 51, 52, 53, 54, 55,
+  56, 57, 97, 98, 99, 100, 101, 102
+];
+
+let uuidData;
+let uuidBatch = 63;
+
+// Implements an RFC 4122 version 4 random UUID.
+// To improve performance, random data is generated in batches
+// large enough to cover 64 UUID's at a time. The uuidData buffer
+// is reused for each, which the first 36 bytes in the buffer
+// used for the serialization.
+function randomUUID() {
+  if (uuidData === undefined) {
+    uuidData = Buffer.allocUnsafe(1060);  // 36 + (16 * 64)
```

Comment:
This logic looks too complex...

Why not use two separate buffers for random data and for the stringification?
Both could be allocated on first call, but separately.